drv/transceivers-server/src/udp.rs

```diff
@@ -663,6 +774,21 @@ impl ServerImpl {
                     .unwrap();
                     error_idx += err_size;
                 } else if result.failure().is_set(module) {
+                    let failure_type = match result.failure_types()[module] {
+                        FpgaI2CFailure::NoModule => HwError::NotPresent,
+                        FpgaI2CFailure::NoPower => HwError::NotPowered,
+                        FpgaI2CFailure::PowerFault => HwError::PowerFault,
+                        FpgaI2CFailure::NotInitialized => {
+                            HwError::NotInitialized
+                        }
+                        FpgaI2CFailure::I2cAddressNack => {
+                            HwError::I2cAddressNack
+                        }
+                        FpgaI2CFailure::I2cByteNack => HwError::I2cByteNack,
+                        // We only mark failures when an error is set, so this
+                        // branch should never match.
+                        FpgaI2CFailure::NoError => panic!(),
```

Comment:
nit: given the comment, this should maybe be
```suggestion
                        FpgaI2CFailure::NoError => unreachable!(),
```